yuyo/repeaters.py

```diff
@@ -0,0 +1,228 @@
+# BSD 3-Clause License
+#
+# Copyright (c) 2020-2021, crazygmr101
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import asyncio
+import datetime
+import inspect
+from typing import Any
+from typing import Awaitable
+from typing import Callable
+from typing import Generic
+from typing import List
+from typing import Optional
+from typing import TypeVar
+
+_function = Callable[..., Awaitable[Any]]
+RepeaterFuncT = TypeVar("RepeaterFuncT", bound=_function)
+
+
+class Repeater(Generic[RepeaterFuncT]):
+    def __init__(
+        self,
+        coro: RepeaterFuncT,
+        *,
+        seconds: Optional[float] = None,
+        minutes: Optional[float] = None,
+        hours: Optional[float] = None,
+        run_count: Optional[int] = None,
+        loop: Optional[asyncio.AbstractEventLoop] = None,
+    ):
+        if not (seconds or minutes or hours):
+            raise  # TODO: What error should this raise?
+        self._delay: float = (hours or 0) * 3600 + (minutes or 0) * 60 + (seconds or 0)
+        self._runs_left = run_count or -1
+        self._event_loop = loop or asyncio.get_event_loop()
+        self._coro = coro
+        self._task: Optional[asyncio.Task[None]] = None
+        self._iter: int = 0
+        self._next_iteration: Optional[datetime.datetime] = None
+        self._before: Optional[RepeaterFuncT] = None
+        self._after: Optional[RepeaterFuncT] = None
+        self.ignored_exceptions: List[type] = []
+        self.fatal_exceptions: List[type] = []
+
```

Comment:
Can we declare a `__call__` which relays self._callback in a similar style to what tanjun's Command objects do